.pre-commit-hooks.yaml

```diff
@@ -95,3 +95,18 @@
   language: script
   entry: hooks/yalc/yalc-check.sh
   files: package\.json
+######################
+# Cspell hook
+- id: cspell
+  name: cspell
+  description: "Checks for spelling mistakes & typos"
+  entry: cspell-cli
+  additional_dependencies: ["--userconfig=./.npmrc","--prefer-online","@turo/cspell@1"]
+  language: node
+  types: [text]
+  args:
+    - --no-must-find-files
+    - --no-progress
+    - --no-summary
+    - --gitignore
+    - --files # this needs to be the last argument so that the paths are passed correctly.
```

Comment:
[side note] 

Not really relevant to the solution. But for the future, these shouldn't be required any more per `cspell` [8.13.3 update](https://github.com/streetsidesoftware/cspell-cli/releases/tag/v8.13.3).

https://team-turo.slack.com/archives/C06QYQV5Y73/p1723576525628529

Reply:
True, but we are technically copying their implementation of the cspell hook here as our own, so essentially as if we had forked their open source hook.

Reply:
So technically, consumers of this new hook of _ours_ don't have to worry about passing these args, because _we've_ passed them under the hood always 😄

Reply:
If we ship this, we should star the cspell pre-commit hook repo, so we can get notified and port over any major changes they make to our repo here, which I think shouldn't be too much trouble